06_puppy/lexual/puppy.go

```diff
@@ -0,0 +1,79 @@
+package main
+
+import (
+	"sync"
+)
+
+type Puppy struct {
+	ID     uint32
+	Breed  string
+	Colour string
+	Value  int
+}
+
+// Storer defines standard CRUD operations for Puppys
+type Storer interface {
+	CreatePuppy(*Puppy)
+	ReadPuppy(id uint32) *Puppy
+	UpdatePuppy(id uint32, puppy *Puppy)
+	DeletePuppy(id uint32) bool
+}
+
+// MapStore ////
+
+type MapStore map[uint32]Puppy
+
```

Comment:
you may consider documenting all exported functions

Reply:
done

Reply:
In addition to this, can you add proper documentation to all your exported types? The `MapStore` type above has a comment which is not directly on the line above, and non descriptive of what that type is. Please update this for all exported types throughout your code.